File: solutions/1768. Merge Strings Alternately/1768.cpp
// Time complexity: O(max(m,n))
// Space complexity: O(m+n)
class Solution {
 public:
  string mergeAlternately(string word1, string word2) {
    const int n = min(word1.length(), word2.length());
    string pref;

    for (int i = 0; i < n; ++i) {
      pref += word1[i];
      pref += word2[i];
    }

    return pref + word1.substr(n) + word2.substr(n);
  }
};

File: solutions/1768. Merge Strings Alternately/1768.py
# Time complexity: O(max(m,n))
# Space complexity: O(m+n)
class Solution:
  def mergeAlternately(self, word1: str, word2: str) -> str:
    return ''.join(a + b for a, b in zip_longest(word1, word2, fillvalue=''))